### ISSUE TITLE
Management ip or float ip to be used in F5 Ingress controller

### ISSUE THREAD

**Issue by zhu20200720** [F5Networks/k8s-bigip-ctlr#1451]
#### Setup Details
CIS Version : 2.1.0
Build: f5networks/k8s-bigip-ctlr:latest
BIGIP Version: Big IP 2.1.0
AS3 Version: 3.x
Agent Mode: AS3 or CCCL
Orchestration: K8S
Orchestration Version: 1.16.0
Pool Mode: Nodeport
Additional Setup details: <Platform/CNI Plugins/ cluster nodes/ etc>

#### Description  
 
We have two data centers, each data center have two F5 appliances which uses the HA active-standby option.

In our implementation, we used float ip in F5 connection URL, i.e. --bigip-url= "float ip of f5 applicance"

For fail over between data center, it is manual.

Per our use case scenario, could anyone suggest that management IP or float IP should be used to connect F5 appliance in Ingress Controller? 

Thanks

#### Steps To Reproduce
1)
2)
3)

#### Diagnostic Information
```buildoutcfg
<Configuration files, error messages, logs>
Note: Sanitize the data. For example, be mindful of IPs, ports, application names and URLs
Note: The following F5 article outlines the information required when opening an issue.
https://support.f5.com/csp/article/K60974137
```

**Comment by cisbotctlr** [comment]
cisbot will assign the issue to one of the devs. 
 @devs, use /jira for internal tracking.

**Comment by veeramalla-f5** [comment]
If I understand your question correct, You are asking us, whether to use Management IP or Floating IP to establish a VXLAN tunnel between F5 and k8s/openshift cluster? Am I correct?

If YES....
I don't think using management is a better option for data plane. However I would request @mdditt2000 to comment on this.

**Comment by zhu20200720** [comment]
@veeramalla-f5 , yes, that was my question. Also we are using AS3 agent but insecure right now.    i.e.   “--insecure=true”, “--agent=as3”. To add a certificate to the CIS trusted SSL certificate store, is it a must to use management ip? Thanks

**Comment by mdditt2000** [comment]
Good morning @zhu20200720 Its not uncommon to to use the self-ip or floating-ip for CIS. If you were going to deploy a dedicated CIS controller per BIG-IP i would use the self-ip of BIG-IP. If you only using a single CIS instance then use the floating-ip. Also note the floating-ip will be used for snat on all Ingress traffic to the Pods. 

Since you are using CIS 2.1 you no longer need agent=as3 in the CIS arguments. AS3 is the default API. Also note "--insecure=true" tells CIS to disable SSL certificate verification required by GOJSON schema validation. You only need to enable  insecure=true if you are using a unsigned or default cert on BIG-IP. This has nothing todo with adding a certificate to CIS trusted SSL certificate store. Here is a document on replacing the BIG-IP system self-signed device certificate https://support.f5.com/csp/article/K42531434

**Comment by zhu20200720** [comment]
Thank you @mdditt2000.  We used **NodePort** type in Ingress Controller, and we also plan to deploy two dedicated Ingress Controller instances for both Active and Standby F5 appliances. For ingress controller configuration
 option, we plan to use F5 management IP, i.e.  “--bigip-url=**management ip**”. I will report the output here in next few days on our testing. 
![bigip-ha](https://user-images.githubusercontent.com/68562586/89575228-794bdf00-d7fb-11ea-895a-6ace40bcb899.png)

**Comment by mdditt2000** [comment]
@zhu20200720 diagram looks good. Let me know how things go. You might also want to enable labels so the master nodes doent should up in the BIG-IP pools members

https://github.com/mdditt2000/kubernetes-1-18/blob/master/kubernetes-faq.md#manage-node-labels

**Comment by zhu20200720** [comment]
For F5 appliance failover testing, we observed the below behavior.

Steps

- Active F5 is brought down
- Standby F5 is made Active
- We make a change on the current F5 UI (deleted a worker node, etc)

We expect Ingress controller should update the F5 and overwrite the changes with the current state of the kubernetes cluster. However we observed that the ingress controller did not overwrite the manual changes we made to the F5. Therefore, I am not sure if we can delete a worker node from F5 UI, and if it can be automatically recovered.

@mdditt2000, I am wondering that if the steps above is a valid testing. I recall we modified health check in F5 UI and it would be overwritten very quickly from the predefined Ingress resource from k8s cluster. 

Thanks

**Comment by howsTricks** [comment]
Would like to elevate interest in this issue, especially from the documentation perspective. Can we please have a recommended design documented when attaching CIS to a HA pair of LTMs?

**Comment by walterk82** [comment]
I think that this discussion is also related to enhancement request https://github.com/F5Networks/k8s-bigip-ctlr/issues/452

I have observed that there seem to be number of questions regrading HA setup along with config sync behaviors.

**Comment by mdditt2000** [comment]
Please see the following user-guides. These guides solve this issue

- Kubernetes User Guide with BIG-IP High Availability (HA) Configuration
https://github.com/mdditt2000/k8s-bigip-ctlr/blob/main/user_guides/k8s-ha/README.md

Multi-Cluster Kubernetes using Container Ingress Services (CIS) and BIG-IP using Flannel
- https://github.com/mdditt2000/k8s-bigip-ctlr/blob/main/user_guides/multi-cluster/user-guide.md

**Comment by mdditt2000** [comment]
Closing issue
